include/seqan3/argument_parser/argument_parser.hpp

```diff
@@ -536,6 +537,9 @@ class argument_parser
     //!\brief Whether the **user** specified to perform the version check (true) or not (false), default unset.
     std::optional<bool> version_check_user_decision;
 
+    //!\brief Befriend seqan3::detail::test_accessor to grant access to version_check_future.
+    friend struct ::seqan3::detail::test_accessor;
```

Comment:
:nail_care: suggestion:
I would propose to put this at the very end of the class. I know we group variables and functions, but this is something very irrelevant to the actual code and should probably not be grouped with "normal" stuff and need not to be seen immediately.

Reply:
I put it near the thing that gets accessed. So that this friend accessor might be seen if we change the something on version_check_future. I propose to put the friend local to the reason why it is there. If a friend has multiple reasons I would put it at the back of the class.

Reply:
Lets put it as a discussion point.

Reply:
We decided that we put the friend `near` the variable.